src/main/java/com/gluonhq/richtext/action/ActionFactory.java

```diff
@@ -0,0 +1,95 @@
+package com.gluonhq.richtext.action;
+
+import com.gluonhq.richtext.RichTextArea;
+import com.gluonhq.richtext.model.TextDecoration;
+import com.gluonhq.richtext.viewmodel.ActionCmdFactory;
+
+public class ActionFactory {
+
+    private static final ActionCmdFactory ACTION_CMD_FACTORY = new ActionCmdFactory();
+    private final RichTextArea control;
+
+    public ActionFactory(RichTextArea control) {
+        this.control = control;
+    }
+
+    private Action undo;
+    public Action undo() {
+        if (undo == null) {
+            undo = new UndoAction(control);
```

Comment:
All these inner classes can be replaced now by directly constructing a `BasicAction`:

```suggestion
            undo = new BasicAction(control, action -> ACTION_CMD_FACTORY.undo());
```

Reply:
done